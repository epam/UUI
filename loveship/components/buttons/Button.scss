// Border style
.style-round {
    &.size-18 { border-radius: 10px; }
    &.size-24 { border-radius: 12px; }
    &.size-30 { border-radius: 15px; }
    &.size-36 { border-radius: 18px; }
    &.size-42 { border-radius: 21px; }
    &.size-48 { border-radius: 24px; }
}