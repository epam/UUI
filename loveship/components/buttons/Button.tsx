import { FillStyle, ControlShape, EpamPrimaryColor } from '../types';
import * as uui from '@epam/uui';
import { createSkinComponent, devLogger } from '@epam/uui-core';
import css from './Button.module.scss';

const defaultSize = '36';

/** Defines component color. */
type ButtonColor = EpamPrimaryColor | 'white' | 'night500' | 'night600' | 'gray' | uui.ButtonProps['color'];

type ButtonMods = {
    /**
     * Defines component color.
     * @default "sky"
     */
    color?: ButtonColor;
    /**
     * Defines component size.
     * @default '36'
     */
    size?: uui.ButtonProps['size'] | '18';
    /**
     * Defines component shape.
     * @default 'square'
     */
    shape?: ControlShape;
    /**
     * Defines component fill style.
     * @default 'solid'
     */
    fill?: FillStyle;
};

const mapFill: Record<FillStyle, uui.ButtonProps['fill']> = {
    solid: 'solid',
    white: 'outline',
    light: 'ghost',
    none: 'none',
};

/** Represents the properties of a Button component. */
export type ButtonProps = uui.ButtonCoreProps & ButtonMods;

function applyButtonMods(mods: ButtonProps) {
    return [
        `uui-size-${mods.size || defaultSize}`,
        css['style-' + (mods.shape || 'square')],
    ];
}

export const Button = /* @__PURE__ */createSkinComponent<uui.ButtonProps, ButtonProps>(
    uui.Button as any, // TODO: remove it when BaseButton inheritance will be reworked
    (props) => {
        if (__DEV__) {
            devLogger.warnAboutDeprecatedPropValue<ButtonProps, 'color'>({
                component: 'Button',
                propName: 'color',
                propValue: props.color,
                propValueUseInstead: 'gray',
                condition: () => ['night500', 'night600'].indexOf(props.color) !== -1,
            });
        }
        return {
            fill: mapFill[props.fill] || mapFill.solid as any,
        };
    },
    applyButtonMods,
);
