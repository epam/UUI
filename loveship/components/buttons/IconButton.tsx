import { IconButton as uuiIconButton, IconButtonProps as UuiIconButtonProps } from '@epam/uui';
import { devLogger, withMods } from '@epam/uui-core';
import { EpamAdditionalColor, EpamPrimaryColor } from '../types';

export interface IconButtonMods {
    /**
     * @default 'night600'
     */
    color?: EpamPrimaryColor | EpamAdditionalColor | 'white' | 'night200' | 'night300' | 'night400' | 'night500' | 'night600';
}

export type IconButtonProps = Omit<UuiIconButtonProps, 'color'> & IconButtonMods;

export const IconButton = withMods<Omit<UuiIconButtonProps, 'color'>, IconButtonMods>(
    uuiIconButton,
    () => [],
    (props) => {
        if (__DEV__) {
            devLogger.warnAboutDeprecatedPropValue<IconButtonProps, 'color'>({
                component: 'IconButton',
                propName: 'color',
                propValue: props.color,
                condition: () => ['night200', 'night300', 'night400'].indexOf(props.color) !== -1,
            });
        }
        return {
            color: props.color ?? 'night600',
        } as IconButtonProps;
    },
);
