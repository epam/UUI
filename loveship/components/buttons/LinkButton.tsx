import { createSkinComponent, devLogger } from '@epam/uui-core';
import * as uui from '@epam/uui';

type LinkButtonColor = 'sky' | 'grass' | 'sun' | 'fire' | 'cobalt' | 'violet' | 'fuchsia' | 'white' | 'night50'
| 'night100' | 'night200' | 'night300' | 'night400' | 'night500' | 'night600' | 'night700' | 'night800' | 'night900'
| uui.LinkButtonProps['color'];

type LinkButtonMods = {
    /**
     * Defines component color.
     * @default 'sky'
     */
    color?: LinkButtonColor;
};
// TODO: leave 3 colors white, blue, gray - more specific colors will be determined later.
/** Represents the properties of a LinkButton component. */
export type LinkButtonProps = uui.LinkButtonCoreProps & LinkButtonMods;

export const LinkButton = /* @__PURE__ */createSkinComponent<uui.LinkButtonProps, LinkButtonProps>(
    uui.LinkButton,
    (props) => {
        if (__DEV__) {
            devLogger.warnAboutDeprecatedPropValue<LinkButtonProps, 'color'>({
                component: 'LinkButton',
                propName: 'color',
                propValue: props.color,
                condition: () => ['grass', 'sun', 'fire', 'cobalt', 'violet', 'fuchsia', 'white', 'night50', 'night200', 'night300', 'night400', 'night500', 'night700', 'night800', 'night900'].indexOf(props.color) !== -1,
            });
        }
        return {
            color: props.color ?? 'sky',
        };
    },
    () => [],
);
