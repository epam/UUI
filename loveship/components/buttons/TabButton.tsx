import css from './TabButton.module.scss';
import * as uui from '@epam/uui';
import { withMods } from '@epam/uui-core';

export interface TabButtonMods extends uui.TabButtonProps {
    theme?: 'light' | 'dark';
}

function applyTabButtonMods(mods: TabButtonMods & uui.TabButtonProps) {
    return [mods.theme === 'dark' && css.themeDark];
}

export interface TabButtonProps extends uui.TabButtonProps, TabButtonMods {}

export const TabButton = withMods<uui.TabButtonProps, TabButtonMods>(uui.TabButton, applyTabButtonMods);
