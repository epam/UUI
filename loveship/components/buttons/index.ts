export * from './Button';
export * from './LinkButton';
export * from './IconButton';
export * from './TabButton';
export * from './VerticalTabButton';
export * from './helper';
