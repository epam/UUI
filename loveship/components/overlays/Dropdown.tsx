import React from 'react';
import { Dropdown as UuiDropdown } from '@epam/uui-components';
import { DropdownProps } from '@epam/uui-core';

export class Dropdown extends React.Component<DropdownProps> {
    render() {
        return <UuiDropdown {...this.props} />;
    }
}
