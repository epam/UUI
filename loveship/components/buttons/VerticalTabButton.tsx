import css from './VerticalTabButton.scss';
import { withMods } from '@epam/uui-core';
import { ButtonProps } from '@epam/uui-components';
import { TabButton, TabButtonMods } from './TabButton';

function applyVerticalTabButtonMods() {
    return [css.root];
}

export const VerticalTabButton = withMods<ButtonProps, TabButtonMods>(TabButton, applyVerticalTabButtonMods);
