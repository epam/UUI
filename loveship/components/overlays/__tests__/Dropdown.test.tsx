import React from 'react';
import renderer from 'react-test-renderer';
import { Dropdown } from '../Dropdown';
import { Button } from '../../buttons';

describe('Dropdown', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<Dropdown renderTarget={props => <Button caption="Test" {...props} />} renderBody={jest.fn()} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with extra props', () => {
        const tree = renderer
            .create(<Dropdown renderTarget={props => <Button caption="test" {...props} />} renderBody={jest.fn()} onClose={jest.fn()} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
