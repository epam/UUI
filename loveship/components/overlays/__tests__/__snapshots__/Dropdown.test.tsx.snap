// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should be rendered correctly 1`] = `
<button
  className="container uui-button-box uui-enabled -clickable button-vars button-primary root size-36 mode-solid uui-theme-loveship size-36 style-square"
  onClick={[Function]}
  tabIndex={0}
  type="button"
>
  <div
    className="uui-caption"
  >
    Test
  </div>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
</button>
`;

exports[`Dropdown should be rendered correctly with extra props 1`] = `
<button
  className="container uui-button-box uui-enabled -clickable button-vars button-primary root size-36 mode-solid uui-theme-loveship size-36 style-square"
  onClick={[Function]}
  tabIndex={0}
  type="button"
>
  <div
    className="uui-caption"
  >
    test
  </div>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
</button>
`;
